front/src/modules/rollingStock/helpers/utils.ts

```diff
@@ -120,84 +106,78 @@ export const getRollingStockEditorDefaultValues = (
-  raise_pantograph_time: data.raisePantographTime,
-  metadata: {
-    detail: data.detail || data.name,
-    family: data.family,
-    grouping: data.grouping,
-    number: data.number,
-    reference: data.reference || data.name,
-    series: data.series,
-    subseries: data.subseries,
-    type: data.type,
-    unit: data.unit,
-  },
-  effort_curves: currentRsEffortCurve,
-  base_power_class: data.basePowerClass,
-});
+): RollingStockForm => {
+  const electric = isElectric(currentRsEffortCurve);
+  return {
+    name: data.name,
+    length: data.length,
+    max_speed: data.maxSpeed / 3.6, // The user enters a value in km/h, which is then interpreted in m/s by the server.
+    startup_time: data.startupTime,
+    startup_acceleration: data.startupAcceleration,
+    comfort_acceleration: data.comfortAcceleration,
+    gamma: {
+      type: 'CONST',
+      value: data.gammaValue,
+    },
+    inertia_coefficient: data.inertiaCoefficient,
+    features: [],
+    mass: data.mass * 1000, // Here we receive a value in ton which will be interpreted in kg by the server.
+    rolling_resistance: {
+      A: data.rollingResistanceA,
+      B: data.rollingResistanceB,
+      C: data.rollingResistanceC,
+      type: 'davis',
+    },
+    loading_gauge: data.loadingGauge,
+    power_restrictions: data.powerRestrictions,
+    energy_sources: [],
+    electrical_power_startup_time: electric ? data.electricalPowerStartupTime : null,
+    raise_pantograph_time: electric ? data.raisePantographTime : null,
+    metadata: {
+      detail: data.detail || data.name,
+      family: data.family,
+      grouping: data.grouping,
+      number: data.number,
+      reference: data.reference || data.name,
+      series: data.series,
+      subseries: data.subseries,
+      type: data.type,
+      unit: data.unit,
+    },
+    effort_curves: currentRsEffortCurve,
+    base_power_class: data.basePowerClass,
+  };
+};
+
+type Conditions = Record<string, (effortCurves: RollingStock['effort_curves'] | null) => boolean>;
 
 export const checkRollingStockFormValidity = (
-  rollingStockForm: RollingStockParametersValues
+  rollingStockForm: RollingStockParametersValues,
+  effortCurves: RollingStock['effort_curves'] | null
 ): { invalidFields: string[]; validRollingStockForm: RollingStockParametersValidValues } => {
-  const invalidFields = [
-    'length',
-    'mass',
-    'maxSpeed',
-    'startupAcceleration',
-    'comfortAcceleration',
-    'startupTime',
-    'gammaValue',
-    'inertiaCoefficient',
-    'rollingResistanceA',
-    'rollingResistanceB',
-    'rollingResistanceC',
-  ].reduce(
-    (result, field) =>
-      !has(rollingStockForm, field) || rollingStockForm[field] === undefined
-        ? [...result, field]
-        : result,
-    [] as string[]
-  );
+  const conditions = RollingStockSchemaProperties.reduce<Conditions>((acc, val) => {
+    if (val.condition) {
+      return { ...acc, [val.title]: val.condition };
+    }
+    return acc;
+  }, {});
```

Comment:
I find this quite abstract/ not readable to get only the dict `{electricalPowerStartupTime: isElectric, raisePantographTime: isElectric}`... I don't have any suggestion, it's just a remark, in case you have an idea to improve this

Reply:
leaving it like this as discussed